File: small-problems/easy-4/short_long_short.rb
# Short Long Short
#
# Problem
#   Receive two strings, determine the longest, return
#     short + long + short
# Example
#   'abc', 'defgh' => 'abcdefghabc'
#   'abcde', 'fgh' => 'fghabcdefgh'
# Data Structure
#   N/A
# Algorithm
#   Use length to determine the longest, and split the
#     into two branches which determine the concatenation
#     order.
# Code
#
def short_long_short(string1, string2)
  first_longer = string1.length > string2.length
  return (string2 + string1 + string2) if first_longer

  (string1 + string2 + string1)
end

File: JS210/exercises/easy-1/short_long_short.js
// Short Long Short

function shortLongShort(str1, str2) {
  const firstLonger = str1.length > str2.length;
  if (firstLonger) {
    return str2 + str1 + str2;
  } else {
    return str1 + str2 + str1;
  }
}

console.log('Test Results');
console.log(shortLongShort('abc', 'defgh') === "abcdefghabc");
console.log(shortLongShort('abcde', 'fgh') === 'fghabcdefgh');
console.log(shortLongShort('', 'xyz') === 'xyz');
